grass7/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py

```diff
@@ -167,14 +167,25 @@ def get_aoi(vector=None):
     if vector:
         args['input'] = vector
 
+    topology = gs.parse_command('v.info', map=vector, flags='t')
+    if int(topology['areas']) < 1:
+        gs.fatal(_("No areas found in AOI map <{}>...").format(vector))
+    elif  int(topology['areas']) > 1:
+        gs.warning(_("More than one areas found in AOI map <{}>. \
+                      Using only the first area...").format(vector))
```

Comment:
I wonder if a modification of the `v.out.ascii` call is needed to really restrict to  "Using only the first area" ?

Reply:
I dont`t think so, cause you limit it to the first geometry here:
```geom = [key for key in geom_dict][0]```

An alternative would be to use a convex hull (v.hull) over input geometries, or loop over polygons...

In principle it would be possible that the first geometry is not an area.
If you want to be 100% sure, check that too:
```if not geom.startswith('POLYGON'):
    gs.fatal(_(''Geometry used as AOI is no area.))```
But that is a corner case I guess...

Reply:
I also thought of convex hull but it is not necessarily the right approach - depends on the use case.

Reply:
Alternative idea: loop over all areas